src/main/java/com/g4s8/ghman/web/TkGitHubAuthorization.java

```diff
@@ -36,43 +38,65 @@
  * @todo #4:30min replace those static loggers with something more OO
  *  (Related with
  *  https://www.yegor256.com/2019/03/19/logging-without-static-logger.html)
- * @todo #4:30min Implement unit tests for TkGitHubAuthorization. Refer
- *  to takes framework Unit testing guide @ https://github.com/yegor256/takes#unit-testing
  */
 final class TkGitHubAuthorization implements Take {
 
+    /**
+     * Code request param.
+     */
+    private static final String P_CODE = "code";
+
     /**
      * Data source.
      */
-    private final DataSource data;
+    private final Users users;
+
+    /**
+     * Authorization function.
+     */
+    private final Func<org.takes.Request, String> auth;
```

Comment:
@olenagerasimova introduce a meaningful abstraction for this. I think the current implementation in the secondary constructor should reside in its own class and there should be a `Fake` implementation for use in the tests.